react/src/components/search-app/SearchBarComponent.jsx

```diff
@@ -0,0 +1,126 @@
+import Autocomplete from 'react-autocomplete';
+import React from 'react';
+
+
+export default class SearchBarComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: props.query,
```

Comment:
If you want the component to handle the data itself, you shouldn't pass it down by props in the first place. Maybe just access `__dataForReact` from here to initialize your state.